x/coinswap/client/cli/tx.go

```diff
@@ -0,0 +1,182 @@
+package cli
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	"github.com/cosmos/cosmos-sdk/x/coinswap/internal/types"
+)
+
+// GetTxCmd returns the transaction commands for this module
+func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
+	coinswapTxCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      "Coinswap transaction subcommands",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	coinswapTxCmd.AddCommand(client.PostCommands(
+		GetCmdAddLiquidity(cdc),
+		GetCmdRemoveLiquidity(cdc))...)
+
+	return coinswapTxCmd
+}
+
+// GetCmdAddLiquidity implements the add liquidity command handler
+func GetCmdAddLiquidity(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "add-liquidity [deposit] [deposit-amount] [min-reward] [deadline]",
+		Args:  cobra.ExactArgs(4),
+		Short: "Add liquidity in the reserve pool",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Add liquidity in the reserve pool for a trading pair.
+			
+Example:
+$ %s tx coinswap add-liquidity dai 1000atom 1000 2 cosmosvaloper1l2rsakp388kuv9k8qzq6lrm9taddae7fpx59wm --from mykey
+`,
+				version.ClientName,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(auth.DefaultTxEncoder(cdc))
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			deposit, err := sdk.ParseCoin(args[0])
+			if err != nil {
+				return err
+			}
+
+			depositAmount, ok := sdk.NewIntFromString(args[1])
+			if !ok {
+				return fmt.Errorf(types.ErrNotPositive(types.DefaultCodespace, "deposit amount provided is not positive").Error())
```

Comment:
The reason I used the same error code is to maintain consistency with the ValidateBasic() method in msgs.go for add and remove liquidity. There, for deposit and withdraw amounts respectively, the above error has been used. 

But yeah since this is the first level of parsing the data, using the same error code won't make sense. I'll look up the common errors that are present in the SDK. If I can't use any of those, I'll add new ones in this module.

Reply:
Since this is the client side, just returning normal golang errors as you suggested.